ansible_collections/arista/avd/roles/eos_cli_config_gen/tasks/main.yml

```diff
@@ -14,53 +14,19 @@
   run_once: true
   register: avd_requirements
 
-- name: Create required output directories if not present
+- name: Generate eos intended configuration and device documentation
+  # TODO this loses the capability to only generate documentation
   tags: [build, provision]
-  ansible.builtin.file:
-    path: "{{ item }}"
-    state: directory
-    mode: 0775
-  loop:
-    - "{{ structured_dir }}"
-    - "{{ documentation_dir }}"
-    - "{{ eos_config_dir }}"
-    - "{{ devices_dir }}"
-  delegate_to: localhost
-  run_once: true
-
-- name: Include device intended structure configuration variables
-  tags: [build, provision]
-  ansible.builtin.include_vars:
-    file: "{{ filename }}"
-  delegate_to: localhost
-  when: structured_config is not defined and lookup('first_found', filename, skip=True, errors='ignore')
-  vars:
-    filename: "{{ structured_dir }}/{{ inventory_hostname }}.{{ avd_structured_config_file_format }}"
-
-- name: Generate eos intended configuration
-  tags: [build, provision]
-  arista.avd.validate_and_template:
-    template: eos-intended-config.j2
-    dest: "{{ eos_config_dir }}/{{ inventory_hostname }}.cfg"
-    mode: 0664
-    schema_id: "{{ role_name }}"
-    conversion_mode: "{{ avd_data_conversion_mode }}"
-    validation_mode: "{{ avd_data_validation_mode }}"
-    #cprofile_file: "config-{{inventory_hostname}}.prof"
-  delegate_to: localhost
-  register: eosconfig
-
-- name: Generate device documentation
-  tags: [build, provision, documentation]
-  arista.avd.validate_and_template:
-    template: "eos-device-documentation.j2"
-    dest: "{{ devices_dir }}/{{ inventory_hostname }}.md"
-    mode: 0664
-    schema_id: "{{ role_name }}"
+  arista.avd.eos_cli_config_gen:
+    structured_config_filename: "{{ structured_config_filename }}"
+    config_filename: "{{ eos_config_dir }}/{{ inventory_hostname }}.cfg"
+    documentation_filename: "{{ devices_dir }}/{{ inventory_hostname }}.md"
+    read_structured_config_from_file: "{{ structured_config is not arista.avd.defined and lookup('first_found', structured_config_filename, skip=True, errors='ignore') is not arista.avd.defined([]) }}"
```

Comment:
Maybe we should simplify this to just be `"{{ structured_config is not arista.avd.defined }}"` and then simply ignore a missing file in the code.

Reply:
[3ca53c1](https://github.com/aristanetworks/avd/pull/4117/commits/3ca53c1d0c6733fae6b0f44719e5d9dbd299b0b6)